files/galaxy/tpv/tool_defaults.yml

```
@@ -42,6 +42,17 @@ tools:
               entity.tpv_tags = entity.tpv_tags.combine(
                   TagSetManager(tags=[pulsar_tag])
               )
+      - if: |
+          # This rule will apply when users have non-exsisting remote resources selected or recorded in Galaxy DB
+          retval = False
+          user_preferences = user.extra_preferences
+          remote_resource_tag = user_preferences.get("distributed_compute|remote_resources", "None")
+          remote_resource_destination = [d.dest_name for d in mapper.destinations.values() if any(d.tpv_dest_tags.filter(tag_value=remote_resource_tag))]
+          if not remote_resource_destination:
+              retval = True
+          retval
```

Anonymous users also need to be taken into consideration.

```suggestion
          if user is not None:
              user_preferences = user.extra_preferences
              remote_resource_tag = user_preferences.get("distributed_compute|remote_resources", "None")
              remote_resource_destination = [d.dest_name for d in mapper.destinations.values() if any(d.tpv_dest_tags.filter(tag_value=remote_resource_tag))]

          user is not None and not remote_resource_destination
```

If you are wondering the same, I tested using `return` statements in TPV rules but they cannot be used.